### PR TITLE
UI: Add retina support and updated icons

### PR DESCRIPTION
**Pull request by DungFu**
OBS Studio currently does not support retina rendering for any of the
images in the app. This adds support for retina pixmap rendering as
well as adding higher resolution versions of each icon that was not
a high enough resolution to support 2x. The icons work when switching
themes and will only render the 2x versions when the device pixel
ratio is greater than or equal to 2.

I also added credits to the readme for images that were already being
used that require credit to the author. If the OBS community has
paid for these images already, I can remove the credits from the
readme. The credit is for the invisible, visible, and gear icons.

## Before
obs-non-retina

## After
obs-retina

### CODE REVIEW

File: UI/locked-checkbox.cpp [hunk 1]
```diff
@@ -7,10 +7,17 @@
 
 LockedCheckBox::LockedCheckBox() : QCheckBox()
 {
-	lockedImage =
-		QPixmap::fromImage(QImage(":/res/images/locked_mask.png"));
-	unlockedImage =
-		QPixmap::fromImage(QImage(":/res/images/unlocked_mask.png"));
+	if (devicePixelRatio() >= 2) {
+		lockedImage =
+			QPixmap::fromImage(QImage(":/res/images/locked_mask@2x.png"));
```
Comment (RytoEX) [inline UI/locked-checkbox.cpp]: Style:  These lines are longer than 80-characters.  You will probably have to line break them differently, like so:
```
		lockedImage = QPixmap::fromImage(QImage(
				":/res/images/locked_mask@2x.png"));
```

File: UI/locked-checkbox.cpp [hunk 2]
```diff
@@ -31,6 +38,6 @@ void LockedCheckBox::paintEvent(QPaintEvent *event)
 			palette().color(foregroundRole()));
 
 	QPainter p(this);
-	p.drawPixmap(0, 0, image.width(), image.height(),
+	p.drawPixmap(0, 0, 16, 16,
```
Comment (RytoEX) [inline UI/locked-checkbox.cpp]: Why are these values being hardcoded here?

File: UI/visibility-checkbox.cpp [hunk 3]
```diff
@@ -7,10 +7,17 @@
 
 VisibilityCheckBox::VisibilityCheckBox() : QCheckBox()
 {
-	checkedImage =
-		QPixmap::fromImage(QImage(":/res/images/visible_mask.png"));
-	uncheckedImage =
-		QPixmap::fromImage(QImage(":/res/images/invisible_mask.png"));
+	if (devicePixelRatio() >= 2) {
+		checkedImage =
+			QPixmap::fromImage(QImage(":/res/images/visible_mask@2x.png"));
```
Comment (RytoEX) [inline UI/visibility-checkbox.cpp]: Style: These lines are longer than 80-characters. You will probably have to line break them differently, like so:
```
		checkedImage = QPixmap::fromImage(QImage(
				":/res/images/visible_mask@2x.png"));
```

File: UI/visibility-checkbox.cpp [hunk 4]
```diff
@@ -31,6 +38,6 @@ void VisibilityCheckBox::paintEvent(QPaintEvent *event)
 			palette().color(foregroundRole()));
 
 	QPainter p(this);
-	p.drawPixmap(0, 0, image.width(), image.height(),
+	p.drawPixmap(0, 0, 16, 16,
```
Comment (RytoEX) [inline UI/visibility-checkbox.cpp]: Why are these values hardcoded?

File: README.rst [hunk 5]
```diff
@@ -47,3 +47,7 @@ Contributing
    you fully understand -- bad advice is worse than no advice.  When it
    comes to something that you don't fully know or understand, please
    defer to the official help or official channels.
+
+Credits
+-------
+ - Icons made by [Freepik](http://www.freepik.com) from [www.flaticon.com](https://www.flaticon.com/) is licensed by [CC 3.0 BY](http://creativecommons.org/licenses/by/3.0/)
```
[the remark this answers is not in the document]
Comment (DungFu) [reply README.rst]: Will fix

File: UI/locked-checkbox.cpp (hunk 1 above)
Comment (DungFu) [reply UI/locked-checkbox.cpp]: Will fix

File: UI/visibility-checkbox.cpp (hunk 3 above)
Comment (DungFu) [reply UI/visibility-checkbox.cpp]: Will fix

**Comment by DungFu** [comment]
@DDRBoxman I could not find a higher resolution source for that icon and figured that if the icon was to be changed, that it should probably happen in a separate PR.

If anyone has the original source for that icon, that would be great and I can add it to this PR.

File: UI/locked-checkbox.cpp (hunk 2 above)
Comment (RytoEX) [reply UI/locked-checkbox.cpp]: How does this affect the icon displays on other operating systems, if at all?

**Comment by DungFu** [comment]
@DDRBoxman I managed to find the source for the output icon and updated the commit with the new file. The screenshots are also updated to reflect the new image.

**Comment by RytoEX** [comment]
@DungFu 
Following up on this, could you please provide links to the icon pack or individual icon pages on Flaticon and/or Freepik's site?  That way we can refer back to it should the need arise.

**Pull request merged**